### ISSUE TITLE
Merge EngineApiRequestHandler and EngineHandler

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#9009]
The EngineApiRequestHandler

https://github.com/paradigmxyz/reth/blob/83cbcb99aebcb8bdcfce549ab2912e033b402f51/crates/engine/tree/src/engine.rs#L151-L154

and EngineHandler

https://github.com/paradigmxyz/reth/blob/83cbcb99aebcb8bdcfce549ab2912e033b402f51/crates/engine/tree/src/engine.rs#L33-L33

is a redundant layer of abstraction if we want to handle execution in a different task then this can be combined into one type.

we expect that the actual tree execution will happen on a dedicated thread that processes everything sequentially:

basically this:

https://github.com/paradigmxyz/reth/blob/83cbcb99aebcb8bdcfce549ab2912e033b402f51/crates/engine/tree/src/engine.rs#L125-L129

is what we need for the actual execution part
